File: solutions/beecrowd/1279/1279.py
import sys


def is_leap(year):
    return (year % 4 == 0 and year % 100 != 0) or year % 400 == 0


def is_huluculu(year):
    return year % 15 == 0


def is_bulukulu(year):
    return year % 55 == 0 and is_leap(year)


first_test_case = True

for line in sys.stdin:
    if first_test_case:
        first_test_case = not first_test_case
    else:
        print()

    year = int(line)

    if is_leap(year):
        print('This is leap year.')
    if is_huluculu(year):
        print('This is huluculu festival year.')
    if is_bulukulu(year):
        print('This is bulukulu festival year.')
    if not is_leap(year) and not is_huluculu(year) and not is_bulukulu(year):
        print('This is an ordinary year.')

File: solutions/beecrowd/1279/1279.clj
(ns main)

(defn leap? [year]
  (or (zero? (mod year 400))
      (and (zero? (mod year 4)) (not (zero? (mod year 100))))))

(defn huluculu? [year]
  (zero? (mod year 15)))

(defn bulukulu? [year]
  (and (zero? (mod year 55))
       (leap? year)))

(defn main []
  (loop [first-test-case true
         line (read-line)]
    (when line
      (when (not first-test-case)
        (println))
      (let [year (bigint line)]
        (when (and (not (leap? year))
                   (not (huluculu? year))
                   (not (bulukulu? year)))
          (println "This is an ordinary year."))
        (when (leap? year)
              (println "This is leap year."))
        (when (huluculu? year)
              (println "This is huluculu festival year."))
        (when (bulukulu? year)
              (println "This is bulukulu festival year.")))
      (recur false (read-line)))))

(main)
